### ISSUE TITLE
❗[Notification] Application deadline approaching

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#9640]
## ✨ Purpose

Remind a user that the deadline for a job application is approaching.

## ❗ Trigger

Send reminders 3 days before deadline.

This requires a cron job which runs every hour. It should look for currently open posters which close in the appropriate window (+/- 30 minutes?). 

## 👀 Audience

Notification should be sent to all users with a draft application for the closing poster.

### In-app string (English and French)

{{opportunity.title}} closes on {{opportunity_due_date_but_actually_the_day_before_EST}}. Continue your application.

FR: Le processus pour {{opportunity.title}} prend fin le 25 avril 2024. Continuer votre soumission.

### Email (English and French)

**Subject:** GC Digital Talent opportunity closes on April 25, 2024

_(already sent to translation by Marc.)_

**Body:**

You started an application to [{{opportunity.title}}](url) which closes in 3 days, at midnight on  {{opportunity_due_date_but_actually_the_day_before_EST}}.
If you’re still interested, we encourage you to [continue your application](url).

_(already sent to translation by Marc.)_

**Footer** (hopefully the same on all notification emails):
GC Digital Talent team

[Get in touch](url) with us to provide feedback or report a problem.

[GC Digital Talent](url) is a recruitment platform for digital jobs in the Government of Canada.

If you no longer want to receive these updates, you can [unsubscribe from notifications](url).
This is an automated message, so please do not reply.

_(already sent to translation by Marc.)_

### Formatting should look something like this... Ideally
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/2494551/82ceb059-72a6-474b-be0b-4e220a867d8d)

### Links

Clicking the in-app notification should send you to continue the draft application. The "continue your application" text in the email should do the same.

In the email body, the "Opportunity Title" should be linked to the job advertisement for that process.

In the email footer "Get in touch" links to the support page https://talent.canada.ca/support .

In the email footer "you can unsubscribe from notifications" links to the new account settings page for managing notifications on the profile.

## 💾 Content

- This should be part of the **application updates** notification family (see #9555)

## ✅ Acceptance Criteria

- [ ] Email template exists in GC Notify
- [ ] Notification class exists in Laravel
- [ ] Notification is triggered by appropriate event and send to appropriate users (including NOT any users who are ignoring this notification's Notification Family) MAKE SURE TO USE TEST ENV
- [ ] Notification type added to schema
- [ ] Notification type rendered appropriately on frontend
- [ ] Tests? (How do we test the cron job?) DEV testing

## ⚙️ Deployment
- Ensure the cron job is running correctly on the server.

## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] Should we make a separate ticket to get cron running on server?
```

**Comment by marc-donofrio** [comment]
Email: [noreply@canada.talent.ca](mailto:noreply@canada.talent.ca)
Subject: Reminder: Opportunity {{opportunity.title}} deadline closing in {{opportunity_remaining_time}}
Hi {{user.firstname}},

Your application to the opportunity {{opportunity.title}} is due in {{opportunity_remaining_time}}.

Details:

Opportunity name - {{opportunity.title}} 

Requestor - {{requester.email}}

Opportunity link - {{opportunity.url}}

This is an automated reminder email from {{portal_name}}

************************
Courriel : [noreply@canada.talent.ca](mailto:noreply@canada.talent.ca)
Objet : Rappel : La date limite de l’opportunité {{opportunity.title}} se termine dans {{opportunity_remaining_time}}
Bonjour {{user.firstname}},

Votre candidature à l’opportunité {{opportunity.title}} doit être soumise dans {{opportunity_remaining_time}}.

Détails :

Nom de l’opportunité - {{opportunity.title}} 

Demandeur - {{requester.email}}

Lien vers l’opportunité - {{opportunity.url}}

Il s’agit d’un e-mail de rappel automatisé de {{portal_name}}

**Comment by tristan-orourke** [comment]
caveat - applicants are responsible for their own submission. Some people will complain they didn't get an email and should get an extension.

**Comment by tristan-orourke** [comment]
Extensions are not available

**Comment by mnigh** [comment]
> Email: [noreply@canada.talent.ca](mailto:noreply@canada.talent.ca)

@marc-donofrio  Is this meant to be `noreply@talent.canada.ca`?

**Comment by gobyrne** [comment]
[FINAL-Opportunity closes notification-FR.docx](https://github.com/GCTC-NTGC/gc-digital-talent/files/15066248/FINAL-Opportunity.closes.notification-FR.docx)

**Comment by gobyrne** [comment]
A couple updates:

1. Added link on words "GC digital talent" in the footer to talent.canada.ca
2. Some changes to French copy that came back. (in-app notification now updated above)

**Comment by gobyrne** [comment]
Actually, I've highlighted all the changed text (from first translation document) in here:

[FINAL-Opportunity closes notification-FR1.1 (002).docx](https://github.com/GCTC-NTGC/gc-digital-talent/files/15154149/FINAL-Opportunity.closes.notification-FR1.1.002.docx)
